Sources/StreamVideoSwiftUI/CallView/VideoRenderer.swift

```diff
@@ -93,21 +94,54 @@ public struct VideoRendererView: UIViewRepresentable {
 
 public class VideoRenderer: RTCMTLVideoView {
     
+    @Injected(\.thermalStateObserver) private var thermalStateObserver
+
     let queue = DispatchQueue(label: "video-track")
     
     weak var track: RTCVideoTrack?
     
     var feedFrames: ((CMSampleBuffer) -> ())?
     
     private var skipNextFrameRendering = true
-    
-    var trackId: String? {
-        self.track?.trackId
+    private var cancellable: AnyCancellable?
+
+    private(set) var preferredFramesPerSecond: Int = UIScreen.main.maximumFramesPerSecond {
+        didSet {
+            metalView?.preferredFramesPerSecond = preferredFramesPerSecond
+            log.debug("🔄 preferredFramesPerSecond was updated to \(preferredFramesPerSecond).")
+        }
     }
-    
+    private lazy var metalView: MTKView? = { subviews.compactMap { $0 as? MTKView }.first }()
```

Comment:
Why not self? How are we sure that `MTKView` is available? Previously we were setting the preferred FPS for `RTCMTLVideoView` (the current view).

Reply:
We cannot use `self.preferredFramesPerSecond` as `RTCMTLVideoView` is a UIView and doesn't have this property available. If you look in the implementation of [RTCMTLVideoView](https://chromium.googlesource.com/external/webrtc/+/fc8c97f950a4fead4a5debe3c838942db1834010/webrtc/sdk/objc/Framework/Classes/Metal/RTCMTLVideoView.m#63) you will see that as long as metal is available the metalView is guaranteed to be there

Reply:
👍 I realized I actually had the same code written before, why am I complaining now 😅